File: javaScript/Gate Three/Quagmire.js


function specifyListLength(listOfNumbers, lengthOfNewList){
	let newListOfNumbers = [];
	let lengthOfListOfNumbers = listOfNumbers.length;
	for (let index = 0; index < lengthOfNewList; index++){
			if (index < lengthOfListOfNumbers){
				newListOfNumbers[index] = listOfNumbers[index];
			}
			else{
				newListOfNumbers[index] = -1;
			}
		}
		return newListOfNumbers;
}

console.log(specifyListLength([1, 2, 3, 4], 6))
console.log(specifyListLength([1, 2, 3, 4], 2))
console.log(specifyListLength([], 4))

File: python/Gate Three/quagmire.py

def specify_list_length(list_of_numbers, length_of_new_list):
    new_list = []
    length_of_list_of_numbers = len(list_of_numbers)
    for index in range (length_of_new_list):
        if index < length_of_list_of_numbers:
            new_list += [list_of_numbers[index]]
        else:
            new_list += [-1]
    return new_list
